### PR TITLE
[PM-8338] New Autofill Settings Components

### PR DESCRIPTION
**Pull request by jprusik**
## 🎟️ Tracking

PM-8338

## 📔 Objective

Create a new Autofill Settings and Excluded Domains components to match Design specs.

## Notes

- Like other extension refresh work, the new components are behind the `extension-refresh` feature flag.
- Existing components have been renamed to use a "v1" naming scheme for easier deprecation in the future
- There are some expected/approved differences from design due to presumptive design system work on the horizon
- Replacements of hyphenated "auto-fill" to "autofill" in the existing view are expected/approved by Design

## 📸 Screenshots

Note: some of the screenshots below were taken before the Autofill on page load warning text update, which looks like this: 
Screenshot 2024-07-19 at 11 45 28 AM

**UX Recording**
https://github.com/user-attachments/assets/2494df29-8889-40e2-9ec4-d2747ef28b7c

Screenshot 2024-07-19 at 10 18 43 AM
Screenshot 2024-07-19 at 10 21 05 AM

**Firefox**
image

**Unknown/unsupported client**
Screenshot 2024-07-19 at 11 39 40 AM

### Supported/Identified Browser
Screenshot 2024-07-19 at 10 19 57 AM
Screenshot 2024-07-19 at 10 20 12 AM
Screenshot 2024-07-19 at 10 20 28 AM
Screenshot 2024-07-19 at 10 21 18 AM
Screenshot 2024-07-19 at 10 21 28 AM
Screenshot 2024-07-19 at 10 21 44 AM
Screenshot 2024-07-19 at 10 26 07 AM

### Unsupported/unidentified browser

Screenshot 2024-07-19 at 11 40 53 AM
Screenshot 2024-07-19 at 11 40 43 AM

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/browser/src/autofill/popup/settings/autofill-v1.component.ts
Comment (jprusik) [inline apps/browser/src/autofill/popup/settings/autofill-v1.component.ts:1]: Largely left unchanged from original code

File: apps/browser/src/autofill/popup/settings/autofill-v1.component.html
Comment (jprusik) [inline apps/browser/src/autofill/popup/settings/autofill-v1.component.html:1]: Largely left unchanged from original code

File: apps/browser/src/autofill/popup/settings/autofill.component.ts [hunk 1]
```diff
@@ -16,72 +29,101 @@ import { I18nService } from "@bitwarden/common/platform/abstractions/i18n.servic
...
+    DisablePasswordManagerUris.Unknown;
+  protected browserShortcutsURI: BrowserShortcutsUri = BrowserShortcutsUris.Unknown;
+  protected browserClientIsUnknown: boolean;
+  enableAutofillOnPageLoad = false;
+  enableInlineMenu = false;
+  enableInlineMenuOnIconSelect = false;
+  autofillOnPageLoadDefault = false;
+  autofillOnPageLoadOptions: { name: string; value: boolean }[];
   enableContextMenuItem = false;
-  enableAutoTotpCopy = false; // TODO: Does it matter if this is set to false or true?
```
Comment (jprusik) [inline apps/browser/src/autofill/popup/settings/autofill.component.ts:39]: To answer this question; not practically, but there is a moment before state is loaded where these values are used

**Review by danielleflinn — APPROVED** [review]
LGTM 🎉

File: libs/common/src/autofill/constants/index.ts [hunk 2]
```diff
@@ -61,3 +61,27 @@ export const AutofillOverlayVisibility = {
   OnButtonClick: 1,
   OnFieldFocus: 2,
 } as const;
+
+export const BrowserClientVendors = {
```
Comment (cagonzalezcs) [inline libs/common/src/autofill/constants/index.ts:65]: 👍

File: apps/browser/src/autofill/popup/settings/autofill.component.ts [hunk 3]
```diff
@@ -16,72 +29,101 @@ import { I18nService } from "@bitwarden/common/platform/abstractions/i18n.servic
...
     ];
     this.clearClipboardOptions = [
-      { name: i18nService.t("never"), value: null },
-      { name: i18nService.t("tenSeconds"), value: 10 },
-      { name: i18nService.t("twentySeconds"), value: 20 },
-      { name: i18nService.t("thirtySeconds"), value: 30 },
-      { name: i18nService.t("oneMinute"), value: 60 },
-      { name: i18nService.t("twoMinutes"), value: 120 },
-      { name: i18nService.t("fiveMinutes"), value: 300 },
+      { name: i18nService.t("never"), value: ClearClipboardDelay.Never },
```
Comment (cagonzalezcs) [inline apps/browser/src/autofill/popup/settings/autofill.component.ts:120]: 🎉

**Review by cagonzalezcs — APPROVED** [review]
Nice work on this, really like how you've cleaned up elements of the original component and removed magic string/number references.

**Pull request merged**